crates/flux-infer/src/fixpoint_encoding.rs

```diff
@@ -211,6 +238,16 @@ impl SortEncodingCtxt {
                 let args = args.iter().map(|s| self.sort_to_fixpoint(s)).collect_vec();
                 fixpoint::Sort::App(fixpoint::SortCtor::Map, args)
             }
+
+            rty::Sort::App(rty::SortCtor::Adt(sort_def), args) if sort_def.is_reflected() => {
```

Comment:
I think this would be cleaner if we didn't care about reflection and only branch on whether this is a struct.